pkg/util/kubernetes/autoscalers/processor.go

```diff
@@ -131,14 +137,54 @@ func (p *Processor) ProcessWPAs(wpa *v1alpha1.WatermarkPodAutoscaler) map[string
 	return externalMetrics
 }
 
+func makeChunks(batch []string) (chunks [][]string) {
+	for i := 0; i < len(batch); i += chunkSize {
+		if i+chunkSize > len(batch) {
+			chunks = append(chunks, batch[i:])
+			break
+		}
+		chunks = append(chunks, batch[i:i+chunkSize])
+	}
+	return chunks
+}
+
```

Comment:
maybe we can speed up a bit this function, something like
```
func makeChunks(batch []string) (chunks [][]string) {
        totalLen := len(batch)
        var nbBatch int
        for {
              if ((nbBatch +1) * chunkSize) >  totalLen {
                     return append(chunks, batch[nbBatch*chunkSize:])
              } else {
                     chunks = append(chunks, batch[nbBatch*chunkSize:(nbBatch +1) * chunkSize)])
              }
              nbBatch++
        }
}
```

Reply:
why is this faster ? I find it harder to read personally

Reply:
true, it is not easier to read.
Here, for `for loop` will be executed only the number of batch.
after I don't think it is performance issue to stay like you did, should not have thousands of metrics.

Reply:
🤞